packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection-list.tsx

```diff
@@ -0,0 +1,113 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import React from 'react';
+import { SavedConnection } from './saved-connection';
+import {
+  Subtitle,
+  Icon,
+  IconButton,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+const savedConnectionListStyles = css({
+  width: '100%',
+  height: '320px',
+  flex: 'none',
+  padding: spacing[3],
+  marginTop: 'auto',
+});
+
+const savedConnectionListPaddingStyles = css({
+  overflowY: 'auto',
+  maxHeight: '100%',
```

Comment:
This max-height is causing the container to not scroll fully because you're basically forcing it to be the full height of the parent, not accounting for the header:

<img width="253" alt="image" src="https://github.com/mongodb-js/compass/assets/5036933/779244c7-2357-460f-84b2-a631149aa475">

<img width="269" alt="image" src="https://github.com/mongodb-js/compass/assets/5036933/e16f0c28-dda6-4607-99ac-b99d7d72274e">

You probably want this container to take all the available space sans header, you can either hardcode the header height here to account for that (a bit clunky, hard to change and maintain) or actually make the parent a flex container to make sure that the list can stretch to all available space, but no more than that.

(the bottom padding of the parent should also probably be moved, otherwise the content weirdly "disappears" in parent when scrolling)